client/src/main/java/org/apache/celeborn/client/write/DataPushQueue.java

```diff
@@ -107,7 +107,7 @@ public ArrayList<PushTask> takePushTasks() throws IOException, InterruptedExcept
               oldCapacity = maxInFlight - pushState.inflightPushes(loc.hostAndPushPort());
               workerCapacity.put(loc.hostAndPushPort(), oldCapacity);
             }
-            workerWaitAttempts.putIfAbsent(loc.hostAndPushPort(), new AtomicInteger(0));
+            workerWaitAttempts.computeIfAbsent(loc.hostAndPushPort(), key -> new AtomicInteger(0));
```

Comment:
Hi @cchung100m @pan3793 , I think computeIfAbsent is superior to putIfAbsent only when the value is heavy to construct/deconstruct. The lambda for computeIfAbsent is not free. So I suggest here we use putIfAbsent, because new an AtomicInteger object should be relatively cheap.

Reply:
Hi @waitinfuture  

Thanks for the feedback and I updated the part you mentioned here.

Reply:
You may interested in this result :)
https://github.com/apache/incubator-celeborn/pull/1657

Reply:
Thanks @pan3793 for the benchmark :)